src/acl/RegexData.cc

```diff
@@ -113,45 +104,28 @@ removeUnnecessaryWildcards(char * t)
     return t;
 }
 
-static bool
-compileRE(std::list<RegexPattern> &curlist, const char * RE, int flags)
+static void
+compileRE(std::list<RegexPattern> &curlist, const SBuf &RE, int flags)
```

Comment:
Now that you have moved the internals away and added compileREs() for all the callers there is no need for this static function. Please inline it to the (one?) caller.

Reply:
> there is no need for this static function

There is a need: If we inline it, we would duplicate its logic. The logic is simple, but there is no good reason to duplicate it. The compiler will probably inline it into both callers (in optimized builds), and that is fine.

P.S. Just to avoid misunderstanding: I did not add compileREs(). I renamed one of the two old compileRE() into compileREs(). I can undo that renaming if you insist, but the code will become more difficult to read/navigate than the current PR code because the context does not always make it clear which of the two same-name functions is being called.

P.P.S. Neither of the two static functions should probably exist as such, but refactoring all this code to properly use class methods is not trivial and is outside this PR scope.

Reply:
I disagree on the utility of de-duplicating one-liners. The argument could be used to de-duplicate the lines calling the wrapper function, and that wrapper function, and that wrapper function...

Your call though.

Reply:
It is not really the number of lines that counts. It's what is _inside_ those line(s) that does. De-duplicating names is usually pointless (and would lead to the endless de-duplication loop you are thinking about). Virtually anything else is fair game! Every case should be evaluated on its own merits, of course, but, when in doubt, we should not duplicate. I have seen numerous times how seemingly trivial de-duplication improvements exposed or fixed serious bugs.

And, as far as this specific duplication is concerned: Adding something to an _ordered_ collection is a known source of bugs -- you and me had to deal with two of emplace_back() bugs in the past few weeks alone! We do not want to duplicate that kind of code!